data/abi/abi_encode.go

```diff
@@ -258,12 +258,14 @@ func tupleEncoding(v Value) ([]byte, error) {
 	}
 
 	// concat everything as the abi encoded bytes
-	head, tail := make([]byte, 0), make([]byte, 0)
-	for i := 0; i < len(v.ABIType.childTypes); i++ {
-		head = append(head, heads[i]...)
-		tail = append(tail, tails[i]...)
+	encoded := make([]byte, 0)
```

Comment:
If the total capacity is known in advance, you can add it as a 3rd argument. This will avoid unnecessary intermediate allocations.
```suggestion
	encoded := make([]byte, 0, headLength+tailCurrLength)
```